src/test/fuzz/util.h

```diff
@@ -470,4 +470,63 @@ void ReadFromStream(FuzzedDataProvider& fuzzed_data_provider, Stream& stream) no
     }
 }
 
+class FuzzedSocket : public Socket
+{
+    FuzzedDataProvider& m_fuzzed_data_provider;
+
+public:
+    explicit FuzzedSocket(FuzzedDataProvider& fuzzed_data_provider) : m_fuzzed_data_provider{fuzzed_data_provider}
+    {
+    }
+
+    ssize_t send(const char* buf, size_t len, int flags) override
+    {
+        if (m_fuzzed_data_provider.ConsumeBool()) {
+            return len;
+        }
+        return m_fuzzed_data_provider.ConsumeIntegralInRange<ssize_t>(-1, len);
+    }
+
+    ssize_t recv(char* buf, size_t len, int flags) override
+    {
+        if (buf == nullptr || len == 0 || m_fuzzed_data_provider.ConsumeBool()) {
+            return m_fuzzed_data_provider.ConsumeBool() ? 0 : -1;
+        }
+        const std::vector<uint8_t> random_bytes = m_fuzzed_data_provider.ConsumeBytes<uint8_t>(len);
+        if (random_bytes.empty()) {
+            return m_fuzzed_data_provider.ConsumeBool() ? 0 : -1;
+        }
+        std::memcpy(buf, random_bytes.data(), random_bytes.size());
+        if (m_fuzzed_data_provider.ConsumeBool()) {
+            if (len > random_bytes.size()) {
+                std::memset(buf + random_bytes.size(), 0, len - random_bytes.size());
+            }
+            return len;
+        }
+        return random_bytes.size();
+    }
+
+#ifdef USE_POLL
+    int poll(short events, int timeout_ms) override
+    {
+        return m_fuzzed_data_provider.ConsumeIntegralInRange<int>(-1, 1);
```

Comment:
Calling this with `-1` as the `min` parameter to `ConsumeIntegralInRange` will result in a very large range since it will be cast to `uint64_t`:
https://github.com/bitcoin/bitcoin/blob/48c1083632687a42ac603d4f241e70616a1d3815/src/test/fuzz/FuzzedDataProvider.h#L87

Maybe an alternative version of `ConsumeIntegralInRange` for just -1,0,1 can be used? Not your fault, it's the header file's fault.

Reply:
Nevermind, godbolt tells me I am wrong.